chore: add dev workshop (#6263)
snapcore/snapcraft: docs/contribute/development.rst

Author: https://canonical-ubuntu-documentation-library--6263.com.readthedocs.build/snapcraft/6263/contribute/development/#develop-with-workshop

If these commands complete without error, your environment is ready.

Develop with Workshop
---------------------

.. admonition:: Work in progress
    :class: caution

    Workshop integration for Snapcraft is under active development. Some
    features may be incomplete or subject to change.

`Workshop <https://canonical-workshop.readthedocs-hosted.com/>`__ is Canonical's
container-based development environment. The Snapcraft repository includes a
Workshop configuration that provisions an isolated Ubuntu 24.04 container with
all tools and dependencies pre-installed. It's an alternative to the local

> environment described in the previous section.

Reviewer: ```suggestion
Local development environments require periodic maintenance and debugging. You can 
instead develop inside a sandbox, where the required development software 
is pre-installed and immutable. Snapcraft provides a development image that's 
controlled with `Workshop <https://canonical-workshop.readthedocs-hosted.com>`__.
```

Users won't know what this is yet, so let's ease them into the understanding.

> When the workshop starts, it automatically runs ``make setup`` inside the

Reviewer: How does the user start it?

Author: Added a bit more, and I've ended this section with a link to the Workshop tutorial.

container. From there, use the pre-configured actions to carry out common
development tasks:

.. code-block:: bash

    workshop run -- setup          # re-run the virtual environment setup
    workshop run -- lint           # run all linters
    workshop run -- format         # run the autoformatters
    workshop run -- test           # run the full test suite
    workshop run -- test-fast      # run the fast test suite
    workshop run -- test-coverage  # run tests with a coverage report
    workshop run -- schema         # regenerate the JSON schema
    workshop run -- clean          # remove temporary build artifacts

Two additional helpers support dependency work.

> ``lock`` runs ``uv lock`` with any arguments you pass, then re-runs

Reviewer: ```suggestion
The ``lock`` helper runs ``uv lock`` with any arguments you pass, then re-runs
```

Author: In the clear light of day I'm changing these to "actions" (including replacing it above).

``make setup``:

.. code-block:: bash

    workshop run -- lock --upgrade-package craft-parts

> ``install-branch`` installs a dependency directly from a Git branch. Pass a

Reviewer: ```suggestion
The ``install-branch`` helper installs a dependency directly from a Git branch. Pass a
```

If these commands complete without error, your environment is ready.

Develop with Workshop
---------------------

.. admonition:: Work in progress
    :class: caution

    Workshop integration for Snapcraft is under active development. Some
    features may be incomplete or subject to change.

Local development environments require periodic maintenance and debugging. You can
instead develop inside a sandbox, where the required development software
is pre-installed and immutable. Snapcraft provides a development image that's
controlled with `Workshop <https://documentation.ubuntu.com/canonical-workshop/>`__.

> With the `Workshop Snap <https://snapcraft.io/workshop>`__ installed, start the
> development environment by running ``workshop launch dev`` from within the Snapcraft
> directory. When the workshop starts, it automatically runs ``make setup`` inside the
> container. From there, use the pre-configured actions to carry out common
> development tasks:

Reviewer 2: I think the inline command is easy to skim over when the rest of the commands are code-blocked:

```suggestion
With the `Workshop Snap <https://snapcraft.io/workshop>`__ installed, start the
development environment from within the Snapcraft directory:

.. code-block:: bash

    workshop launch dev

When the workshop starts, it automatically runs ``make setup`` inside the
container. From there, use the pre-configured actions to carry out common
development tasks:
```

Reviewer: I like this. :+1:
